### PR TITLE
feat(az): support az pwsh module

### PR DESCRIPTION
**Pull request by tradiff**
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

Change the az segment to support the Azure PowerShell Module

Closes #569

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: docs/docs/segment-environment.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ make sure to do this after importing `go-my-posh` and you're good to go.
 function Set-EnvVar {
     $env:POSH=$(Get-Date)
 }
-New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVar' -Scope Global
+New-Alias -Name 'Set-PoshContext' -Value 'Set-EnvVar' -Scope Global -Force
```
Comment (tradiff) [inline docs/docs/segment-environment.md:32]: `-Force` allows the user to run `.$profile` to reload their profile without an error message.

File: src/segment_az.go [hunk 2]
```diff
@@ -36,24 +36,47 @@ func (a *az) init(props *properties, env environmentInfo) {
 }
 
 func (a *az) enabled() bool {
```
Comment (tradiff) [inline src/segment_az.go:38]: I decided to make the EnvVars have priority over the CLI for two reasons:
1. The CLI is slow and this segment has always been slow on my computer.  So we do the faster operation first, and early exit if it was successful.
2. The EnvVars is something the user is opting in to by adding code to the $profile.  To me, the act of doing that signals that the user wants the EnvVars to be used.

File: docs/docs/segment-az.md [hunk 3]
```diff
@@ -31,3 +31,27 @@ Display the currently active Azure subscription information.
...
+## Az PowerShell Module
+
+By default, the az segment will retrieve subscription information for the [Azure CLI][az-cli], not the [Az PowerShell Module][pwsh-module].
+If you use the Az PowerShell Module, add the code below to your PowerShell profile. This will cause
+the az segment to show information relevant to the Az PowerShell module.
+
+```pwsh
+function Set-EnvVar {
+    try {
+        $subscription = Get-AzContext | Select-Object -ExpandProperty "Subscription" | Select-Object "Name", "Id"
```
Comment (JanDeDobbeleer) [inline docs/docs/segment-az.md]: we could preload this just like we do for `posh-git`. If that cmdlet is available, add those env vars.

Comment (tradiff) [reply docs/docs/segment-az.md]: done!

File: src/init/omp.ps1 [hunk 4]
```diff
@@ -23,6 +23,19 @@ function global:Set-PoshGitStatus {
     }
 }
 
+function global:Set-AzSubscription {
```
Comment (JanDeDobbeleer) [inline src/init/omp.ps1]: Looking at that we now also have `Set-PoshGitStatus` I would maybe rename that one to `Initialize-ModuleSupport` and add this logic there as well. That way we have 1 spot for those items and expand as we see fit.

Comment (tradiff) [reply src/init/omp.ps1]: Sure, I can do that

Comment (tradiff) [reply src/init/omp.ps1]: done!

**Comment by JanDeDobbeleer** [comment]
@TravisTX if you rebase LGTM

**Pull request merged**